Feature: allow user to disable info fields with flag --disable

ktsuench opened the pull request:
I implemented the feature using the package `Strum` to work with the new `InfoFields` enum. The package allows iteration over enums and creates static str for each value in the enum essentially allowing us to create a vector of selected values from the enum (less code). In this case, I hid a enum value `UnrecognizedField` as the end user should not see that value in the output of possible values. That value is used to filter out the default value which is an empty string allowing all info fields to be shown.

Use:
Overrides showing the dominant language ascii logo
```
--disable <disable_field>...         Disable fields to show
                                     Possible values: ["project", "head", "version", "created", "languages",
                                     "authors", "last_change", "repo", "commits", "lines_of_code", "size",
                                     "license"]`
```

Examples:
1. Flag used with no value provided, defaults to showing all info

![image](https://user-images.githubusercontent.com/6343135/66736778-28b46100-ee38-11e9-9367-53f7cc69e701.png)

2. Flag used with unrecognized field, CLI errors out

![image](https://user-images.githubusercontent.com/6343135/66736822-4bdf1080-ee38-11e9-8aac-e611f9ae8f75.png)

3. Flag used with recognized field, shows all info except that field

![image](https://user-images.githubusercontent.com/6343135/66736883-76c96480-ee38-11e9-9bee-866fdf1637c5.png)

4. Flag used with multiple recognized fields in different cases, shows all info except those fields

![image](https://user-images.githubusercontent.com/6343135/66736935-9bbdd780-ee38-11e9-92dd-65e9f63e372d.png)

5. Flag used with languages, shows all info except languages and ASCII logo still shows

![image](https://user-images.githubusercontent.com/6343135/66737024-d6277480-ee38-11e9-944c-52dace6882cc.png)

6. Flag used with recognized and unrecognized fields, CLI errors out

![image](https://user-images.githubusercontent.com/6343135/66737078-ffe09b80-ee38-11e9-899f-32296ef8f3eb.png)

7. Flag used with all recognized fields, no info shows

![image](https://user-images.githubusercontent.com/6343135/66737204-551cad00-ee39-11e9-9bad-0797b057f49e.png)


Addresses issue #73.

ktsuench commented:
Actually, I realized I could simplify the code a bit more by removing the default value for `--disable` which fully utilizes the conditional assignment to `disable_fields` variable. Since there's no use for `UnrecognizedField` value in the enum anymore, I can just use `EnumVariantNames` which implements `variants` method on the enum and returns an array of `&'static str` of values from the enum. The behaviour is still the same as above except case 1 now errors out as shown:

![image](https://user-images.githubusercontent.com/6343135/66738055-92823a00-ee3b-11e9-9d2a-4e8f93a85de6.png)

o2sh commented:
Very Impressive @ktsuench,

Still, I think It would be a better user experience, if the program would not fail if the info name(s) inputed by the user is either missing or not included in the possible values.

* Case 1: `onefetch --disable created head not_a_valid_input repo` --> doesn't fail and shows all info names except head created and repo
* Case 2: `onefetch --disable` -->  doesn't fail and shows all info names
* Case 3: `onefetch --disable not_a_valid_input ` -->  doesn't fail and shows all info names

What do you think ?

ktsuench commented:
I agree with you on that. Using my first solution we could do that by removing the set of possible values and just have unrecognized values fall back to `UnrecognizedField` which will get filtered out.

ktsuench commented:
Reverted my last commit and updated to allow any value but values are checked afterwards.

1. Flag used with no value provided, defaults to showing all info

![image](https://user-images.githubusercontent.com/6343135/66760252-b6f80980-ee6f-11e9-8078-4df228048ea0.png)

2. Flag used with unrecognized field, defaults to showing all info

![image](https://user-images.githubusercontent.com/6343135/66760292-cd05ca00-ee6f-11e9-9998-530395fe0808.png)

3. Flag used with recognized and unrecognized fields, ignores unrecognized fields and hides recognized ones

![image](https://user-images.githubusercontent.com/6343135/66760341-e575e480-ee6f-11e9-9268-71812b9640c9.png)

spenserblack commented on src/main.rs:
```diff
@@ -55,35 +57,43 @@ impl fmt::Display for Info {
+                "HEAD: ".color(color).bold(),
+                self.current_commit
+            )?;
+        }
```
Running `Vec::contains` multiple times is kind of computationally intensive, as each call will iterate through until finding a match. See source code [here](https://doc.rust-lang.org/src/core/slice/mod.rs.html#1282-1286) and [here](https://doc.rust-lang.org/src/core/slice/mod.rs.html#5557-5561).

Iterating through once and keeping track `bool` values might be better.
Example:
```rust
let mut no_project = false;
// etc.
for field in self.disable_fields.iter() {
    match field {
        InfoFields::Project => no_project = true,
        // etc.
    }
}

if !no_project {
    // write project
}
```

ktsuench replied:
I decided to create a struct of bools just to make it easier to track.
